src/modules/webpack-publish.test.ts

```diff
@@ -0,0 +1,123 @@
+import { WebpackRestoreOptions } from "../interfaces/webpack-restore-options";
+import { NodeCI } from "./node-ci";
+import { NodeClean } from "./node-clean";
+import { NodeRestore } from "./node-restore";
+import { WebpackPublish } from "./webpack-publish";
+
+describe("webpack-publish", () => {
+    describe("restore", () => {
+        beforeEach(() => {
+            jest.spyOn(NodeClean, "run").mockImplementation();
+            jest.spyOn(NodeClean, "run").mockImplementation();
+            jest.spyOn(NodeCI, "run").mockImplementation();
+        });
+        test("when ci is null, then NodeCI is not called ", () => {
```

Comment:
Suggestion: Recommend merging this and the following test and pass `null` and `false` as a param given the tests are identical. While the spy is setup in two different places between the two tests, I believe it should just be setup in the arrange, like it is in this test.

Reply:
@dylanjustice You should be able to pull out variables and assign them in the `beforeEach` itself - 

```ts
let nodeCleanRunSpy;
    beforeEach(() => {
        nodeCleanSpy = jest.spyOn(NodeClean, "run").mockImplementation();
    }
    // ...
```

I think I saw it in here at some point, was it not working? Of course, if you had these spies setup in the `beforeEach` hook and wanted to override that behavior (mock it with a different implementation), you could reassign it in the arrangement of your test-

```ts
nodeCleanRunSpy = jest.spyOn(NodeClean, "run").mockImplementation(() => { console.log("Hello world!" });
```